### ISSUE TITLE
Performance troubles / Chrome crashes

### ISSUE THREAD

**Issue by Nebel54** [phase2/particle#504]
Hello there,

we have integrated around 60 components in our particle theme. Right now I cannot use patternlab anymore because as soon as I click on one menu items chrome starts saying "processing request…" in the status bar and stops working with the only output being "Aw snap! Something went wrong while disabling this page". As there is no error message, I run out of debugging options.

So far I tried disabling component groups one by one. When I disable half of them patternlab works again, but I cannot locate a single component that is responsible for the crash. So I guess I scratched a memory limit or something else, but cannot find out if it is on client side or if some payload is missing from the server.

Is there a global way to enable verbosity of the CLI output so I can get an idea of what has been requested or to see if a specific process stops working during the output generation?

Any tips are highly appreciated!

**Comment by Nebel54** [comment]
I think I found the reason why the browser crashed: The generated CSS+JS files where each about 57 MB. I think we have some nasty recursive includes in our code, which we still need to locate.

Here are the steps I used to debug the problem.

**Step 1:**
In package.json I increased the memory_limit to be able to compile the project and added --profile params to the command: 
`"build:pl": "NODE_OPTIONS=--max-old-space-size=3072 NODE_ENV='production' webpack --config ./apps/pl/webpack.pl.prod.js  --profile --progress --json > stats.json",`

**Step 2:**
run `npm run build:pl`

**Step 3:**
Remove unnecessary output from the beginning of stats.json

**Step 4:**
Upload the generated stats.json to https://webpack.github.io/analyse/ -> This tool provides detailed insights into the build process, all modules and eventually the asset sizes :)

**Comment by Nebel54** [comment]
Just another quick follow-up (for future-me and google):

The reason Chrome crashed was simply that I printed stuff in _variables.scss. This file gets embedded in every component and is not supposed to print anything. That lead to printing classes for icon fonts 80-120 times.

But even then the CSS file size was over 45MB because source maps where stored inline in CSS - this has been fixed in #381 but I was working with an older version of particle.
